node/pkg/driver/node_utils.go

```diff
@@ -580,6 +590,39 @@ func (d NodeUtils) GetBlockVolumeStats(mpathDevice string) (VolumeStatistics, er
 	return volumeStats, nil
 }
 
+func (d NodeUtils) IsVolumePathMatchesVolumeId(volumeId string, volumePath string) (bool, error) {
+	logger.Infof("IsVolumePathMatchesVolumeId: Searching matching volume id for volume path: [%s] ", volumePath)
+	volumeUuid := d.GetVolumeUuid(volumeId)
+	volumeUids := d.osDeviceConnectivityHelper.GetVolumeIdVariations(volumeUuid)
+	mpathdOutput, err := d.osDeviceConnectivityHelper.GetMpathdOutputByVolumeIds(volumeUids)
+	if err != nil {
+		return false, err
+	}
+
+	mpathDeviceName, err := d.osDeviceConnectivityHelper.GetMpathDeviceName(volumePath)
+	if err != nil {
+		return false, err
+	}
+
+	volumeIdByVolumePath, err := d.osDeviceConnectivityHelper.GetMatchingVolumeIdToMpathName(mpathdOutput, mpathDeviceName, volumeUids)
+	if err != nil {
+		return false, err
+	}
+	logger.Infof("IsVolumePathMatchesVolumeId: found volume id [%s] for volume path [%s] ", volumeId, volumePath)
```

Comment:
why `volumeId`? isn't `volumeIdByVolumePath` the volume id found for the volume path?

Reply:
done